docs/index.adoc

```diff
@@ -61,46 +63,161 @@ ____
+The requirement for the port number is, that it must be unique on the machine where the FMU binary is executed.
+Typically, a certain range of ports is reserved for this purpose.
+The FMU importer is responsible for checking if any conflicts of the defined IP addresses and port numbers occur in the context of the simulated system.
 
-It is recommended to start the XCP service during `fmi3EnterInitializationMode` (or earlier) and shut down during `fmi3Terminate` (or later) if the FMU has no explicit power-up signal to simplify user interactions between simulator and MCD tool.
+Sometimes it is necessary for the FMU importer to override the default IP address and/or port number which was assigned by the FMU exporter.
+Therefore, the FMU model description shall contain two structural parameters which are used by the XCP slave during the configuration phase to setup the connection for the XCP protocol, see <<XCP Slave Implementation>>.
 
+.XCP connection parameters
+[[figure-xcp-connection-parameters]]
+----
+    IPAddress
+        Description:  "IP address or host name of the machine where the FMU binary is executed"
+        Type:         String
+        Causality:    structuralParameter
+        Variability:  fixed
+        Start:        "127.0.0.1"
+
+    XcpPortNumber
+        Description:  "Port number where the XCP slave listens for XCP protocol commands"
+        Type:         UInt16
+        Causality:    structuralParameter
+        Variability:  fixed
+        Start:        <in the range of 32768 to 39999>
+----
+
+The consumer of an FMU is responsible for keeping all occurrences of the IP address and port number consistent, either by manual adjustment of the FMU model description and the A2L file, or with tool support.
+
+
+=== XCP Slave Implementation
+
+The XCP service shall be started during `fmi3ExitConfigurationMode` and shut down during `fmi3Terminate` if the FMU has no explicit power-up signal to simplify user interactions between simulator and MCD tool.
 If the FMU contains a virtual ECU with power-up control (K15), all build in OS and Basic Software services (including XCP) should follow normal power-up protocol.
 
-_[The A2L/XCP standards allow to measure variables synchronous to different time- or angular-based events._
-_This requires to invoke the XCP service with a so-called raster ID in the thread which is executed for an event._
-_Calibration is typically performed in a background thread._
-_Refer to the A2L/XCP standards for more information.]_
+The XCP slave shall use the structural parameters `IPAddress` and `XcpPortNumber` (see <<figure-xcp-connection-parameters>>) to set up the communication connection for the XCP protocol.
+After leaving `Configuration Mode` the XCP slave shall be responsive for XCP commands.
+Thus, it is possible for the XCP master to perform calibration during the `Instantiated` state of the FMU, for example, to set parameters before entering `Initialization Mode`.
```

Comment:
Aha - this kind of answers my question from above... but I think this requires some discussion:
Since XCP master and importer/FMU state machines are not synchronised, the XCP master cannot know when Instantiated is exited.
If such fine control is required in the XCP protocoll/measurement, the A2L should contain proper `instantiated` events corresponding to the XCP slave in the FMU answering appropriately.
I think this standard would have to be stretched quite a bit to enforce/allow such assurances.
Or: What am I missing?